collect_app/src/main/java/org/odk/collect/android/widgets/interfaces/GeoButtonClickListener.java

```diff
@@ -0,0 +1,13 @@
+package org.odk.collect.android.widgets.interfaces;
+
+import android.content.Context;
+import android.os.Bundle;
+
+import org.javarosa.core.model.FormIndex;
+import org.odk.collect.android.utilities.PermissionUtils;
+import org.odk.collect.android.widgets.utilities.WaitingForDataRegistry;
+
+public interface GeoButtonClickListener {
```

Comment:
I think this interface should really be called the `GeoDataRequester` as that's what the widgets use it for. They do that on a button click but that's a detail of the widget.  It should also hide anything to do with the geo activity classes from the widget. The widget doesn't care how the geo data is acquired right? It just wants to get it in `setBinaryData` somehow.

 That being the case it probably makes sense to have different methods like `requestGeoPoint`, `requestGeoShape` etc and then have the implementation control what activity it launches in each case. The implementation should also take responsibility for building the `Bundle` and the `requestCode`. 

Given all that I think the implementation should probably be called something like `ActivityGeoDataRequester` as it's an activity based implementation of the request geo data.

Reply:
I renamed `GeoDataRequester` class back to `GeoWidgetUtils`, as I thought it also has methods to convert answer string to geo data etc. I renamed the interface as `ActivirtyGeoDatarequester`, which has methods `requesGeoPoint`, `requestGeoShape`, `requestGeoTrace` and `requestGeoIntent`. Also, I am testing the interface separately in the `GeoWidgetUtilsTest`.

Reply:
Might be worth reading my comment again as I don't think that's what I was getting at. I don't think the `GeoWidgetUtils` should implement the `GeoDataRequester` interface (it should just be a non construtable utils class with helper methods) and I don't think the interface should be named `ActivityGeoDataRequester`, that was a suggested name for the implementation. Also, if it wasn't clear I was thinking you would get rid of `requestGeoIntent` and just have the three methods `requesGeoPoint`, `requestGeoShape`, `requestGeoTrace` which would all be `void`.

Very happy to discuss more if it's helpful!